types/src/state_store/state_value.rs

```diff
@@ -157,27 +274,19 @@ impl StateValue {
         self,
         f: F,
     ) -> anyhow::Result<StateValue> {
-        Ok(StateValue::new_impl(match self.inner {
-            StateValueInner::V0(data) => StateValueInner::V0(f(data)?),
-            StateValueInner::WithMetadata { data, metadata } => StateValueInner::WithMetadata {
-                data: f(data)?,
-                metadata,
-            },
-        }))
+        Ok(Self::new_with_metadata(
+            f(self.inner.data)?,
+            self.inner.metadata,
+        ))
     }
 
-    pub fn into_metadata(self) -> Option<StateValueMetadata> {
-        match self.inner {
-            StateValueInner::V0(_) => None,
-            StateValueInner::WithMetadata { metadata, .. } => Some(metadata),
-        }
+    pub fn into_metadata(self) -> StateValueMetadata {
+        self.inner.metadata
     }
 
-    pub fn into(self) -> (Option<StateValueMetadata>, Bytes) {
-        match self.inner {
-            StateValueInner::V0(bytes) => (None, bytes),
-            StateValueInner::WithMetadata { data, metadata } => (Some(metadata), data),
-        }
+    pub fn into_parts(self) -> (StateValueMetadata, Bytes) {
```

Comment:
nit: this is typically called unpack?

Reply:
I think we call it into_inner() throughout the code :)

Reply:
... Just changed to `unpack`.. 

we have more `unpack` and `into_inner` than `into_parts`, to be fair.